core_impl/src/main/java/com/google/instrumentation/trace/SpanData.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.trace;
+
+import com.google.common.base.MoreObjects;
+import com.google.instrumentation.common.Timestamp;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/** Immutable representation of all data collected by the {@link Span} class. */
+@Immutable
+public final class SpanData {
```

Comment:
suppose this would be an input object zipkin folks could convert from similar to how we do in htrace.. 
https://github.com/apache/incubator-htrace/tree/master/htrace-zipkin